Library/Homebrew/dev-cmd/update-python-resources.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  module_function
+
+  def update_python_resources_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `update-python-resources` [<options>] <formula>
+
+        Update versions for PyPI resource blocks in <formula>.
+      EOS
+      switch "-p", "--print-only",
+             description: "Print the updated resource blocks instead of changing the formula."
+      named :formula
+    end
+  end
+
+  def update_python_resources
+    update_python_resources_args.parse
+
+    formula = args.formulae.first
+
+    pypi_resources = formula.resources.select do |resource|
+      resource.url.match?(%r{^https?://files.pythonhosted.org/packages/})
+    end
+
+    odie "No PyPI resources found in \"#{formula.name}\"" if pypi_resources.empty?
+
+    odie "\"python@3.8\" must be installed" unless Formula["python@3.8"].any_version_installed?
+
+    unless File.exist?("#{HOMEBREW_PREFIX}/bin/virtualenv")
+      odie "Please install \"virtualenv\" by running `pip3 install virtualenv`"
+    end
+
+    Mktemp.new("poet-env").run do
+      ohai "Temporarily installing \"#{formula.name}\"" unless args.print_only?
+
+      # Create virtualenv and install package and poet
+      Utils.safe_popen_read "#{HOMEBREW_PREFIX}/bin/virtualenv", "venv"
+      Utils.safe_popen_read "venv/bin/pip", "install", formula.name, "homebrew-pypi-poet"
```

Comment:
The formula name almost always but doesn't always exactly match the PyPI name, eg `gimme-aws-creds.rb`. The most reliable way to get the correct PyPI name is to grab the value between the final `/` and the final `-` of the primary `pythonhosted` url.

Reply:
Good idea. One question: are there instances where the primary url is not a `pythonhosted` url but it still has `pythonhosted` urls?

I know `ansible` does this but it's also a special case for other reasons...

I think I could use this when possible and fallback to using `formula.name` when the primary url isn't from `pythonhosted`. I also need to add some more logic to give a nice error if the package couldn't be installed.